haystack_experimental/super_components/indexers/document_indexer.py

```diff
@@ -2,88 +2,94 @@
 
-from typing import Any, Dict
+from typing import Any, Dict, List, Optional
 
 from haystack import Pipeline, component, default_from_dict, default_to_dict
-from haystack.components.embedders import (
-    AzureOpenAIDocumentEmbedder,
-    HuggingFaceAPIDocumentEmbedder,
-    OpenAIDocumentEmbedder,
-    SentenceTransformersDocumentEmbedder,
-)
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder
 from haystack.components.writers import DocumentWriter
-from haystack.core.component import Component
 from haystack.core.serialization import (
     DeserializationError,
     component_from_dict,
     component_to_dict,
     import_class_by_name,
 )
+from haystack.document_stores.in_memory import InMemoryDocumentStore
 from haystack.document_stores.types import DocumentStore, DuplicatePolicy
 
 from haystack_experimental.core.super_component import SuperComponentBase
 
-KNOWN_DOCUMENT_EMBEDDERS = (
-    HuggingFaceAPIDocumentEmbedder,
-    SentenceTransformersDocumentEmbedder,
-    OpenAIDocumentEmbedder,
-    AzureOpenAIDocumentEmbedder,
-)
-
-
-class InvalidEmbedderError(ValueError):
-    """Raised when a DocumentIndexer receives an invalid embedder parameter."""
+DEFAULT_EMBEDDING_MODEL = "sentence-transformers/all-mpnet-base-v2"
 
 
 @component
 class DocumentIndexer(SuperComponentBase):
     """
-    A document indexer that takes a list of documents and indexes them using the specified embedder.
+    A document indexer that takes a list of documents and indexes them using the specified model.
 
     Usage:
-    ```python
-    from haystack import Document
-    from haystack.components.embedders import SentenceTransformersDocumentEmbedder
-    from haystack.document_stores.in_memory import InMemoryDocumentStore
-    from haystack.document_stores.types import DuplicatePolicy
-
-    doc = Document(content="I love pizza!")
-
-    indexer = DocumentIndexer(
-        embedder=SentenceTransformersDocumentEmbedder(),
-        document_store=InMemoryDocumentStore(),
-        duplicate_policy=DuplicatePolicy.OVERWRITE,
-    )
-
-    indexer.warm_up()
-
-    result = indexer.run(documents=[doc])
-    print(result)
 
-    # {'documents_written': 1}
+    ```python
+    >>> from haystack import Document
+    >>> doc = Document(content="I love pizza!")
+    >>> indexer = DocumentIndexer()
+    >>> indexer.warm_up()
+    >>> result = indexer.run(documents=[doc])
+    >>> print(result)
+    {'documents_written': 1}
+    >>> indexer.pipeline.get_component("writer").document_store.count_documents()
+    1
     ```
     """
 
     def __init__(
         self,
-        embedder: Component,
-        document_store: DocumentStore,
+        model: Optional[str] = None,
+        prefix: str = "",
+        suffix: str = "",
+        batch_size: int = 32,
+        embedding_separator: str = "\n",
+        meta_fields_to_embed: Optional[List[str]] = None,
+        document_store: Optional[DocumentStore] = None,
```

Comment:
Let's not pass a document_store here. This is just for showcasing, let's use InMemoryDocumentStore hard-coded­ for now.